### PR TITLE
Update sdk & cleanup apis

### PR DESCRIPTION
**Pull request by rampr**
1. With this change, the API calls have been updated to the latest version of onesignal
2. The SDK has been cleaned up and all the data is sent to the application and all logic has to be taken care of by the application itself

---

This change is [Reviewable](https://reviewable.io/reviews/hashcube/onesignal/1)

### CODE REVIEW

**Comment by rampr** [comment]
Review status: 0 of 5 files reviewed at latest revision, 1 unresolved discussion.

---

_[android/manifest.xsl, line 12 [r1]](https://reviewable.io:443/reviews/hashcube/onesignal/1#-KQJzTUlxdHganJQrmu3:-KQJzTUlxdHganJQrmu4:b-lyx7nv) ([raw file](https://github.com/hashcube/onesignal/blob/74d38ce6df81af23d7c703759f9cf14c8e7a201f/android/manifest.xsl#L12)):_

> ``` XSLT
>   </xsl:template>
> 
>   <xsl:template match="meta-data[@android:name='googleProjectNo']">
> ```

was defined twice

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/hashcube/onesignal/1)_

**Comment by bijoshtj** [comment]
_[android/manifest.xml, line 15 [r2]](https://reviewable.io:443/reviews/hashcube/onesignal/1#-KQK3VsZ3zW0NcZ97r2s:-KQK3VsZ3zW0NcZ97r2t:byz6rx0) ([raw file](https://github.com/hashcube/onesignal/blob/217c36f314e4704d050a6e7f2108dbbeafb226a9/android/manifest.xml#L15)):_

> ``` XML
> 
> 
> <uses-permission android:name="com.sec.android.provider.badge.permission.READ"/>
> ```

different devices needs different permissions?

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/hashcube/onesignal/1)_

**Comment by rampr** [comment]
Review status: 0 of 5 files reviewed at latest revision, 2 unresolved discussions.

---

_[android/manifest.xml, line 15 [r2]](https://reviewable.io:443/reviews/hashcube/onesignal/1#-KQK3VsZ3zW0NcZ97r2s:-KQK4uuBeE5ix94CgONc:b-q5ruc6) ([raw file](https://github.com/hashcube/onesignal/blob/217c36f314e4704d050a6e7f2108dbbeafb226a9/android/manifest.xml#L15)):_
<details><summary>

<i>Previously, Bijoshtj (Bijosh T J) wrote…</i></summary>



> different devices needs different permissions?
> </details>
> This is to the read the badges counter

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/hashcube/onesignal/1)_

**Comment by rohanhc** [comment]
:lgtm:

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/hashcube/onesignal/1#-:-KQKg4_SRzu-RAiCRh0T:bnfp4nl)_

File: android/manifest.xml [hunk 1]
```diff
@@ -1,16 +1,40 @@
 <!--START_PLUGINS_MANIFEST-->
 <!-- Onesignal -->
+<permission android:name="c2d_message"
+            android:protectionLevel="signature" />
+<uses-permission android:name="packageNamePermissionC2D" />
```
Comment (jishnu7) [inline android/manifest.xml:5]: this doesn't look like a valid permission.

**Comment by rampr** [comment]
Review status: 0 of 5 files reviewed at latest revision, 3 unresolved discussions.

---

_[android/manifest.xml, line 5 [r2]](https://reviewable.io:443/reviews/hashcube/onesignal/1#-KQJzDE9rvC-H6nOr_DV-r2-5:-KQPHbkO55uD2GXnYb6y:b-8emlo9) ([raw file](https://github.com/hashcube/onesignal/blob/217c36f314e4704d050a6e7f2108dbbeafb226a9/android/manifest.xml#L5)):_
<details><summary>

<i>Previously, jishnu7 (Jishnu Mohan) wrote…</i></summary>



> this doesn't look like a valid permission.
> </details>
> The stylesheet(manifest.xsl) will replace these with the right values

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/hashcube/onesignal/1)_

**Comment by rampr** [comment]
Review status: 0 of 5 files reviewed at latest revision, 4 unresolved discussions.

---

_[android/manifest.xsl, line 22 [r2]](https://reviewable.io:443/reviews/hashcube/onesignal/1#-KQPI-iRNURpp6jC1YLo:-KQPI-iRNURpp6jC1YLp:bjfusb1) ([raw file](https://github.com/hashcube/onesignal/blob/217c36f314e4704d050a6e7f2108dbbeafb226a9/android/manifest.xsl#L22)):_

> ``` XSLT
>   <xsl:template match="uses-permission[@android:name='packageNamePermissionC2D']">
>     <uses-permission android:name="{$packageName}.permission.C2D_MESSAGE"/>
>   </xsl:template>
> ```

@jishnu7 It will be replaced with this

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/hashcube/onesignal/1)_

**Comment by jishnu7** [comment]
LGTM

**Comment by bijoshtj** [comment]
:lgtm:

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/hashcube/onesignal/1#-:-KQQIAifwWAGGYx4lU4p:bnfp4nl)_

**Comment by bijoshtj** [comment]
Reviewed 4 of 5 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/hashcube/onesignal/1)_

File: android/OnesignalPlugin.java [hunk 2]
```diff
@@ -263,51 +170,47 @@ public boolean consumeOnBackPressed() {
...
-
-        EventQueue.pushEvent(new onesignalNotificationOpened(
-                           onesignal_data.toString()));
-        data_to_send.put("last_notification_opened_on", current_time.toString());
+        os_data.put("notification_id", payload.notificationId);
+        os_data.put("title", payload.title);
+        os_data.put("body", payload.body);
+        additional_data = payload.additionalData;
+        if (additional_data != null) {
+          os_data.put("additional_data", additional_data.toString());
```
Comment (sksethi25) [inline android/OnesignalPlugin.java:207]: should add is_active i think

Comment (rampr) [reply android/OnesignalPlugin.java:207]: done, change the variable to is_foreground

**Comment by sksethi25** [comment]
Approved :)

**Pull request merged**